recipes/opencv/2.x/conanfile.py

```diff
@@ -71,176 +56,145 @@ def config_options(self):
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("zlib/1.2.12")
-        if self.options.with_jpeg:
-            self.requires("libjpeg/9d")
+        self.requires("zlib/1.2.13")
+        if self.options.with_jpeg == "libjpeg":
+            self.requires("libjpeg/9e")
+        elif self.options.with_jpeg == "libjpeg-turbo":
+            self.requires("libjpeg-turbo/2.1.4")
+        elif self.options.with_jpeg == "mozjpeg":
+            self.requires("mozjpeg/4.1.1")
         if self.options.with_png:
-            self.requires("libpng/1.6.37")
+            self.requires("libpng/1.6.39")
         if self.options.with_jasper:
-            self.requires("jasper/2.0.33")
+            self.requires("jasper/4.0.0")
         if self.options.with_openexr:
+            # opencv 2.x doesn't support openexr >= 3
```

Comment:
would it be worth sticking a validate check in to let the user know?

Reply:
yes but in another PR if you don't mind

Reply:
👍